internal/provider/resources/install/ssh/aws_install.go

```diff
@@ -0,0 +1,219 @@
+package installssh
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/p0-security/terraform-provider-p0/internal"
+	installresources "github.com/p0-security/terraform-provider-p0/internal/provider/resources/install"
+	installaws "github.com/p0-security/terraform-provider-p0/internal/provider/resources/install/aws"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &AwsSshIamWrite{}
+var _ resource.ResourceWithConfigure = &AwsSshIamWrite{}
+var _ resource.ResourceWithImportState = &AwsSshIamWrite{}
+
+type AwsSshIamWrite struct {
+	installer *installresources.Install
+}
+
+type awsSshIamWriteModel struct {
+	AccountId types.String `tfsdk:"account_id" json:"accountId,omitempty"`
+	GroupKey  types.String `tfsdk:"group_key" json:"groupKey,omitempty"`
+	State     types.String `tfsdk:"state" json:"state,omitempty"`
+	Label     types.String `tfsdk:"label" json:"label,omitempty"`
+}
+
+type awsSshIamWriteJson struct {
```

Comment:
It feels like it's a bigger headache trying to get the type system to work properly if we use one struct.

Golang doesn't like getting unknown or nil values for state/local parameters. And it can't marshal strings to the other base types

<img width="238" alt="image" src="https://github.com/p0-security/terraform-provider-p0/assets/12995427/26ba60aa-f790-49c4-8298-65e970123d91">

<img width="574" alt="image" src="https://github.com/p0-security/terraform-provider-p0/assets/12995427/46243a82-fef3-4437-97dc-0dd9aca5c25e">